packages/flutter/lib/src/widgets/snapshot_widget.dart

```diff
@@ -0,0 +1,269 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui;
+
+import 'package:flutter/rendering.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+import 'media_query.dart';
+
+/// Controls how the [SnapshotWidget] paints its child via the [SnapshotWidgetController].
+enum SnapshotMode {
+  /// the child is snapshotted, but only if all descendants can be snapshotted.
+  ///
+  /// If there is a platform view in the children of a raster widget, the
+  /// snapshot will not be used and the child will be rendered as normal.
+  permissive,
+
+  /// the child is snapshotted, but only if all descendants can be snapshotted.
+  ///
+  /// An error is thrown if an unrasterizable view is encountered. This setting is
+  /// the default state of the [SnapshotWidgetController].
+  normal,
+
+  /// The child is snapshotted and any child platform views are ignored.
+  ///
+  /// This mode can be useful if there is a platform view descendant that does
+  /// not need to be included in the snapshot.
+  forced,
+}
+
+/// A controller for the [SnapshotWidget] that controls when the child image is displayed
+/// and when to regenerated the child image.
+///
+/// When the value of [enabled] is true, the [SnapshotWidget] will paint the child
+/// widgets based on the [SnapshotMode] of the snapshot widget.
+///
+/// To force [SnapshotWidget] to recreate the child image, call [clear].
```

Comment:
document when this ChangeNotifier notifies its listeners.

Reply:
Done